Composer/packages/client/src/pages/design/CommandBar.tsx

```diff
@@ -0,0 +1,178 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/** @jsx jsx */
+import { jsx } from '@emotion/core';
+import React, { useMemo, useCallback } from 'react';
+import formatMessage from 'format-message';
+import get from 'lodash/get';
+import { getEditorAPI } from '@bfc/shared';
+import { useRecoilValue } from 'recoil';
+
+import { Toolbar, IToolbarItem } from '../../components/Toolbar';
+import { createDiagnosticsPageUrl, navigateTo } from '../../utils/navigation';
+import {
+  visualEditorSelectionState,
+  dispatcherState,
+  rootBotProjectIdSelector,
+  currentDialogState,
+} from '../../recoilModel';
+import { undoFunctionState } from '../../recoilModel/undo/history';
+import { undoStatusSelectorFamily } from '../../recoilModel/selectors/undo';
+import { DiagnosticsHeader } from '../../components/DiagnosticsHeader';
+import TelemetryClient from '../../telemetry/TelemetryClient';
+
+type CommandBarProps = { dialogId?: string; projectId: string };
+
+const CommandBar: React.FC<CommandBarProps> = ({ dialogId, projectId }) => {
+  const currentDialog = useRecoilValue(currentDialogState({ dialogId, projectId }));
+  const { undo, redo } = useRecoilValue(undoFunctionState(projectId));
+  const rootProjectId = useRecoilValue(rootBotProjectIdSelector) ?? projectId;
+  const visualEditorSelection = useRecoilValue(visualEditorSelectionState);
+  const [canUndo, canRedo] = useRecoilValue(undoStatusSelectorFamily(projectId));
+
+  const { onboardingAddCoachMarkRef } = useRecoilValue(dispatcherState);
+
+  const { actionSelected, showDisableBtn, showEnableBtn } = useMemo(() => {
+    const actionSelected = Array.isArray(visualEditorSelection) && visualEditorSelection.length > 0;
+    if (!actionSelected) {
+      return { actionSelected: false, showDisableBtn: false, showEnableBtn: false };
+    }
+    const selectedActions = visualEditorSelection.map((id) => get(currentDialog?.content, id));
+    const showDisableBtn = selectedActions.some((x) => get(x, 'disabled') !== true);
+    const showEnableBtn = selectedActions.some((x) => get(x, 'disabled') === true);
+
+    return { actionSelected, showDisableBtn, showEnableBtn };
+  }, [visualEditorSelection, currentDialog?.content]);
+
+  const EditorAPI = getEditorAPI();
+
+  const toolbarItems: IToolbarItem[] = [
```

Comment:
Can this be memoized?

Reply:
Or do u think its better to get this PR in to get the structure and we can optimize in successive PR's